dev/com.ibm.ws.security.kerberos.auth/src/com/ibm/ws/security/kerberos/auth/internal/LRUCache.java

```diff
@@ -0,0 +1,91 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.security.kerberos.auth.internal;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+
+import javax.security.auth.Subject;
+import javax.security.auth.kerberos.KerberosPrincipal;
+import javax.security.auth.kerberos.KerberosTicket;
+
+import org.ietf.jgss.GSSCredential;
+import org.ietf.jgss.GSSException;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.websphere.ras.annotation.Trivial;
+
+@Trivial
+public class LRUCache {
+
+    private static final TraceComponent tc = Tr.register(LRUCache.class);
+
+    private final Map<KerberosPrincipal, Subject> cache = new LinkedHashMap<KerberosPrincipal, Subject>() {
```

Comment:
We need to make sure the KerberosPrinicpal is a fully qualify Kerberos principal name userName@KerberosRealmName

Reply:
@utle it will automatically add the default realm name if it is not specified in the principal string, this is the main reason I'm using `KerberosPrinicpal` here instead of just `String`. I ran this test to show that the default realm name is automatically added:
```java
KerberosPrincipal p1 = new KerberosPrincipal("dbuser");
KerberosPrincipal p2 = new KerberosPrincipal("dbuser@EXAMPLE.COM");

System.out.println("equal? " + p1.equals(p2)); // shows both principals are equal
```